tests/src/pcm_out_test.cc

```diff
@@ -192,13 +192,13 @@ TEST_F(PcmOutMmapTest, Write) {
         buffer[i] = static_cast<char>(i);
     }
 
-    int written_frames = 0;
+    int res = 0;
     unsigned int frames = pcm_bytes_to_frames(pcm_object, buffer_size);
     pcm_start(pcm_object);
     auto start = std::chrono::steady_clock::now();
     for (uint32_t i = 0; i < write_count; ++i) {
-        written_frames = pcm_mmap_write(pcm_object, buffer.get(), buffer_size);
-        ASSERT_EQ(written_frames, frames);
+        res = pcm_mmap_write(pcm_object, buffer.get(), buffer_size);
```

Comment:
Why is this changed to zero?

Reply:
That is because of backward compatibility. In the downstream projects, the callers always treat non-zero return value of pcm_mmap_read/write as error. We should keep the behavior of these old APIs and let callers want to get real written frames in return value use the pcm_readi/writei. 

Note 1: pcm_readi/writei can deal with pcm with MMAP flag.
Note 2: I also mark the pcm_mmap_read/write as deprecated functions.